playbook/app/pb_kits/playbook/pb_typeahead/docs/_typeahead_with_highlight.jsx

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable react/no-danger */
+/* eslint-disable react/no-multi-comp */
+/* @flow */
+
+import React, { useState } from 'react'
+import { components, OptionProps } from 'react-select'
+
+import {
+  Avatar,
+  Body,
+  Flex,
+  FlexItem,
+  Title,
+  Typeahead,
+} from '../..'
+
+const USERS = [
+  {
+    name: "Wade Winningham",
+    title: "Nitro Principal Developer",
+    territory: "PHL",
+  },
+  {
+    name: "Jason Cypret",
+    title: "Vice President of User Experience",
+    territory: "PHL",
+  },
+  {
+    name: "Stephen Marshall",
+    title: "Senior User Experience Engineer",
+    territory: "PHL",
+  },
+  {
+    name: "Jasper Furniss",
+    title: "Senior User Experience Engineer",
+    territory: "PHL",
+  }
+];
+
+const TypeaheadWithPillsHighlight = (props) => {
+  const [selectedUser, setSelectedUser] = useState()
+
+  const formatOptionLabel = ({name, territory, title}, {inputValue}) => {
+    const highlighted = (text: string) => {
+      if (!inputValue.length) return text
+      return text.replace(
+        new RegExp(inputValue, 'gi'),
+        highlighted => `<mark>${highlighted}</mark>`
+      )
+    }
+    return (
+      <Flex>
+        <FlexItem>
+          <Avatar
+              marginRight="sm"
+              name={name}
+              size="sm"
+          />
+        </FlexItem>
+        <FlexItem>
+          <Title size={4}><span dangerouslySetInnerHTML={{ __html: highlighted(name) }} /></Title>
+          <Body color="light">
+            <span dangerouslySetInnerHTML={{ __html: highlighted(title) }} />{" • "}
+            {territory}
+          </Body>
+        </FlexItem>
+      </Flex>
+    )
+  }
+
+  const customComponents = {
+    Option: (props: OptionProps) => (
+      <components.Option {...props}/>
+    ),
+    SingleValue: ({ data }: any) => (
+      <span>{data.name}</span>
+    )
+  }
+
+  return (
+    <React.Fragment>
+      <Typeahead
+          components={customComponents}
+          formatOptionLabel={formatOptionLabel}
+          getOptionLabel={(option) => option.name}
+          getOptionValue={(option) => option.title}
```

Comment:
These are our search fields. Above, inside `formatOptionLabel` we receive whatever value we have formatted here.